Description: Fix a malloc() off-by-one error.
Author: Peter Pentchev <roam@ringlet.net>
Forwarded: yes
Last-Update: 2014-11-26

--- a/mbuffer.c
+++ b/mbuffer.c
@@ -1855,7 +1855,7 @@
 
 	/* setup logging prefix */
 	progname = basename(argv0);
-	PrefixLen = strlen(progname) + 2;
+	PrefixLen = strlen(progname) + 3;
 	Prefix = malloc(PrefixLen);
 	(void) strcpy(Prefix,progname);
 	Prefix[PrefixLen - 2] = ':';
